adminSiteServer/apiRoutes/mdims.ts

```diff
@@ -119,3 +185,269 @@ export async function handlePatchMultiDim(
+        MultiDimRedirectsTableName
+    )
+        .select("id")
+        .where("source", source)
+        .first()
+
+    if (existingRedirect) {
+        throw new JsonError(
+            `A redirect with source '${source}' already exists`,
+            400
+        )
+    }
+
+    const targetDescription = buildRedirectTargetDescription(
+        multiDim,
+        viewConfigId ?? null
+    )
+
+    if (source === `/grapher/${multiDim.slug}`) {
+        throw new JsonError(
+            "Creating this redirect would redirect to the same page.",
+            400
+        )
+    }
+
+    const sourceSlug = Url.fromURL(source).slug
+
+    if (await redirectWithSourceExists(trx, source)) {
+        throw new JsonError(
+            `A site redirect with source '${source}' already exists`,
+            400
+        )
+    }
+
+    const existingChartSlugRedirect = await trx<{ id: number }>(
+        ChartSlugRedirectsTableName
+    )
+        .select("id")
+        .where("slug", sourceSlug)
+        .first()
+
+    if (existingChartSlugRedirect) {
+        throw new JsonError(
+            `A chart slug redirect with source '${sourceSlug}' already exists`,
+            400
+        )
+    }
+
+    const publishedChartExists = source.startsWith("/grapher")
+        ? await trx("chart_configs")
+              .leftJoin("charts", "charts.configId", "chart_configs.id")
+              .select(trx.raw("1"))
+              .where("chart_configs.slug", sourceSlug)
+              .whereNotNull("charts.publishedAt")
+              .first()
+        : false
+
+    const publishedExplorerExists = source.startsWith("/explorers")
+        ? await trx(ExplorersTableName)
+              .select(trx.raw("1"))
+              .where({ slug: sourceSlug, isPublished: true })
+              .first()
+        : false
+
+    const publishedMultiDimExists = source.startsWith("/grapher")
+        ? await trx(MultiDimDataPagesTableName)
+              .select(trx.raw("1"))
+              .where({ slug: sourceSlug, published: true })
+              .whereNot("id", multiDimId)
+              .first()
+        : false
+
+    let publishedConflict: string | null = null
+    if (publishedChartExists) {
+        publishedConflict = "a published chart"
+    } else if (publishedExplorerExists) {
+        publishedConflict = "a published explorer"
+    } else if (publishedMultiDimExists) {
+        publishedConflict = "a published multi-dim"
+    }
+
+    if (publishedConflict) {
+        throw new JsonError(
+            `Source '${source}' conflicts with ${publishedConflict}`,
+            400
+        )
+    }
+
+    if (sourceSlug) {
+        const sourceMultiDim = await getMultiDimDataPageBySlug(
+            trx,
+            sourceSlug,
+            { onlyPublished: false }
+        )
+
+        if (sourceMultiDim && sourceMultiDim.id !== multiDimId) {
+            const existingTargetRedirect = await trx<{
+                source: string
```

Comment:
I don't get how we can ever get there. This requires both:
- the sourceSlug to be matching an unpublished multi-dim page (since matching a published multi-dim page is handled above)
- and the (unpublished) multi-dim page to be the target of a redirect, which is not possible in the current state of the UI (since a multi-dim redirect can only be created for published multi-dim - but maybe this is relaxed in the next PR?)